### PR TITLE
bf: do not include current UploadIdMarker in list

### PR DESCRIPTION
**Pull request by philipyoo**
for in_memory only
listMultipartUpload should not list current marker in the
listing. Previously, it would include the marker as the
first item in the list

### CODE REVIEW

File: lib/storage/metadata/in_memory/bucket_utilities.js [hunk 1]
```diff
@@ -6,9 +6,9 @@ function markerFilterMPU(allMarkers, array) {
         // as or alphabetically after the uploadId of the item,
         // eliminate the item.
         if (uploadIdMarker && keyMarker === array[i].key) {
```
Comment (jonathan-gramain) [inline lib/storage/metadata/in_memory/bucket_utilities.js]: The logic of the whole function seems quite convoluted for what it does, what about going to a simpler:

```javascript
const firstUnfilteredIndex = array.findIndex(
    item => (uploadIdMarker && item.key === keyMarker ?
             item.uploadId > uploadIdMarker :
             item.key > keyMarker));
    return firstUnfilteredIndex !== -1 ? array.slice(firstUnfilteredIndex) : array;
```

Comment (philipyoo) [reply lib/storage/metadata/in_memory/bucket_utilities.js]: nice!

**Comment by philipyoo** [comment]
@ironman-machine
RUN_BACKBEAT_CRR_TESTS=true
try

**Comment by philipyoo** [comment]
@ironman-machine
RUN_BACKBEAT_CRR_TESTS=true
try

**Comment by philipyoo** [comment]
@ironman-machine
RUN_BACKBEAT_CRR_TESTS=true
try

**Pull request merged**